src/hooks/useAllServices.ts

```diff
@@ -95,105 +105,119 @@ const filterAllServicesSections = (allServicesLinks: AllServicesSection[], filte
+        return {
+          ...item,
+          links: findNavItems(item.links, availableLinks),
+        };
+      } else if (isAllServicesLink(item)) {
+        return item;
+      }
+      const [bundle, nav] = (item as string).split('.');
+      const currBundle = availableLinks.find(({ id }) => id === bundle)?.items || {};
+      return Object.values(currBundle).find(({ id }) => id === nav);
+    })
+    .filter(Boolean) as (AllServicesLink | AllServicesGroup)[];
+
 const useAllServices = () => {
-  const [{ availableLinks, ready, error }, setState] = useState<{
+  const [{ ready, error, availableSections }, setState] = useState<{
     error: boolean;
     ready: boolean;
-    availableLinks: NavItem[];
+    availableLinks: BundleNav[];
+    availableSections: AllServicesSection[];
   }>({
     ready: false,
     availableLinks: [],
     error: false,
+    availableSections: [],
   });
   const isMounted = useRef(false);
   const [filterValue, setFilterValue] = useState('');
   // TODO: move constant once the AppFilter is fully replaced
   const bundles = requiredBundles;
+  const fetchNavitation = useCallback(
+    () =>
+      Promise.all(
+        bundles.map((fragment) =>
+          axios
+            .get<BundleNav>(`${getChromeStaticPathname('navigation')}/${fragment}-navigation.json?ts=${Date.now()}`)
+            .catch(() => axios.get<BundleNavigation>(`${isBeta() ? '/beta' : ''}/config/chrome/${fragment}-navigation.json?ts=${Date.now()}`))
+            .then(handleBundleResponse)
+            .catch((err) => {
+              console.error('Unable to load appfilter bundle', err, fragment);
+              return [];
+            })
+        )
+      ),
+    []
+  );
+  const fetchSections = useCallback(
+    async (): Promise<
+      (Omit<AllServicesSection, 'links'> & {
+        links: (string | AllServicesLink | AllServicesGroup)[];
+      })[]
+    > => (await axios.get(`${getChromeStaticPathname('services')}/services.json`)).data,
+    []
+  );
   useEffect(() => {
     isMounted.current = true;
-    Promise.all(
-      bundles.map((fragment) =>
-        axios
-          .get<BundleNavigation>(`${getChromeStaticPathname('navigation')}/${fragment}-navigation.json?ts=${Date.now()}`)
-          // fallback static CSC for EE env
-          .catch(() => axios.get<BundleNavigation>(`${isBeta() ? '/beta' : ''}/config/chrome/${fragment}-navigation.json?ts=${Date.now()}`))
-          .then(handleBundleResponse)
-          .catch((err) => {
-            console.error('Unable to load appfilter bundle', err, fragment);
-            return [];
-          })
-      )
-    ).then((bundleItems) => {
+    const setNavigation = async () => {
+      const bundleItems = await fetchNavitation();
+      const sections = await fetchSections();
       if (isMounted.current) {
-        const availableLinks = parseBundlesToObject(bundleItems.flat());
+        const availableLinks = (bundleItems as BundleNav[]).map((bundle) => {
```

Comment:
If you update the `fetchNavitation` you won't need the type cast here as the return type will always be `Promise<BundleNav[]>` instead of `Promise<never[] | BundleNav[]>`.